### ISSUE TITLE
Don't raise reportFunctionMemberAccess for `__text_signature__` and `__signature__` attributes of functions

### ISSUE THREAD

**Issue by jonathandung** [microsoft/pyright#11435]
Type: <b>Bug</b>

The `__text_signature__` and `__signature__` attributes, especially when set manually, are likely meant to alter the help output of the function, pass stubtest, etc.. Though somewhat hacky, they are perfectly legitimate attributes to set on functions (I get why accessing them would create diagnostics, but maybe assignment to them should not error).

Extension version: 2026.2.101
VS Code version: Code 1.119.0 (8b640eef5a6c6089c029249d48efa5c99adf7d51, 2026-05-05T11:23:50-07:00)
OS version: Windows_NT x64 10.0.26200
Modes:

<details>
<summary>System Added Logs</summary>

```
2026-05-10 22:12:31.329 [info] Client: Pylance client (2026.2.101) started with python extension (2026.5.2026050801)
2026-05-10 22:12:32.230 [info] (9620): Server root directory: file:///c%3A/Users/Jonathan%20Dung/.vscode/extensions/ms-python.vscode-pylance-2026.2.101/dist
2026-05-10 22:12:32.237 [info] (9620): Pylance language server 2026.2.101 (pyright version 1.1.409, commit 34c556b7) starting
2026-05-10 22:12:32.261 [info] (9620): Starting service instance "asyncutils" for workspace "c:\Users\Jonathan Dung\asyncutils"
2026-05-10 22:12:33.670 [info] (9620): Loading background bundle file:///c%3A/Users/Jonathan%20Dung/.vscode/extensions/ms-python.vscode-pylance-2026.2.101/dist/copilot-background.bundle.js into file:///c%3A/Users/Jonathan%20Dung/asyncutils...
2026-05-10 22:12:33.800 [info] (9620): Starting Mcp server id:60395 ...
2026-05-10 22:12:33.825 [info] (9620): FG(asyncutils): Loading pyproject.toml file at c:\Users\Jonathan Dung\asyncutils\pyproject.toml
2026-05-10 22:12:33.831 [info] (9620): FG(asyncutils): Setting environmentName for service "asyncutils": "3.14.0 (global)"
2026-05-10 22:12:33.831 [info] (9620): FG(asyncutils): Setting pythonPath for service "asyncutils": "C:\Program Files\Python314\python.exe"
2026-05-10 22:12:33.832 [info] (9620): FG(asyncutils): No include entries specified; assuming c:\Users\Jonathan Dung\asyncutils
2026-05-10 22:12:33.833 [info] (9620): FG(asyncutils): Auto-excluding **/node_modules
2026-05-10 22:12:33.833 [info] (9620): FG(asyncutils): Auto-excluding **/__pycache__
2026-05-10 22:12:33.833 [info] (9620): FG(asyncutils): Auto-excluding **/.*
2026-05-10 22:12:33.948 [info] (9620): FG(asyncutils): Assuming Python version 3.14.0.final.0
2026-05-10 22:12:34.153 [info] (9620): [Editable Installs] Starting dynamic pth file resolution for Python: C:\Program Files\Python314\python.exe
2026-05-10 22:12:34.227 [info] (9620): [Editable Installs] No sandbox available for Python C:\Program Files\Python314\python.exe. Dynamic pth file resolution requires Python 3.13 or later.
2026-05-10 22:12:34.230 [info] (9620): FG(asyncutils): Found 124 source files
2026-05-10 22:12:34.823 [info] (9620): PQ(asyncutils): Priority queue background worker(2) root directory: file:///c%3A/Users/Jonathan%20Dung/.vscode/extensions/ms-python.vscode-pylance-2026.2.101/dist
2026-05-10 22:12:34.823 [info] (9620): PQ(asyncutils): Priority queue background worker(2) started
2026-05-10 22:12:34.823 [info] (9620): Settings have not changed. Skipping update.
2026-05-10 22:12:43.516 [info] (9620): FG(asyncutils): IDX(asyncutils): Indexer background runner(3) root directory: file:///c%3A/Users/Jonathan%20Dung/.vscode/extensions/ms-python.vscode-pylance-2026.2.101/dist (index)
2026-05-10 22:12:43.516 [info] (9620): FG(asyncutils): IDX(asyncutils): Indexing(3) started
2026-05-10 22:12:45.365 [info] (9620): FG(asyncutils): IDX(asyncutils): scanned(3) 574 files over 1 exec env
2026-05-10 22:12:45.768 [info] (9620): FG(asyncutils): IDX(asyncutils): indexed(3) 574 files over 1 exec env
2026-05-10 22:12:45.877 [info] (9620): FG(asyncutils): IDX(asyncutils): Indexing finished(3).

```
</details><details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz (8 x 2419)|
|GPU Status|2d_canvas: enabled<br>GPU0: VENDOR= 0x8086, DEVICE=0x9a49 [Intel(R) Iris(R) Xe Graphics], DRIVER_VENDOR=Intel, DRIVER_VERSION=32.0.101.6737 *ACTIVE*<br>GPU1: VENDOR= 0x1414, DEVICE=0x008c [Microsoft Basic Render Driver], DRIVER_VERSION=10.0.26100.8328<br>Machine model name: <br>Machine model version: <br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|7.85GB (2.11GB free)|
|Process Argv|--crash-reporter-id 9453c078-702b-4b0c-be82-802577994672|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
binariesv615:30325510
nativeloc1:31344060
dwcopilot:31170013
dwoutputs:31242946
copilot_t_ci:31333650
e5gg6876:31282496
pythonrdcb7:31342333
6518g693:31463988
aj953862:31281341
6abeh943:31336334
envsdeactivate2:31505458
cloudbuttont:31379625
42190218_ostrepl:31403339
ec5jj548:31422691
cp_cls_t_966_ss:31454198
inlinechat_v2_hd992725:31505030
4je02754:31466945
8hhj4413:31478653
38bie571_auto:31429954
cp_cls_t_1081:31454832
conptydll_true:31498968
ia-use-proxy-models-svc:31452481
e9c30283:31461165
test_treatment2:31471001
46204921:31447328
th_e6b84190:31511298
idci7584:31464702
e3e4d672:31494082
ei9d7968:31496641
nes-extended-on:31455476
chat:31457767
8hig5102:31480529
89g7j272:31507493
7e187181:31503455
i2gc6536:31499202
ghj88844:31499326
ec1eg703:31491645
ddid_c:31478207
getcmakediagnosticson:31489824
ja75b849:31510500
7df3h592:31509626
pro_large_t:31499376
cp_cls_c_1082:31504161
logging_enabled_new:31498466
j0d79568:31499440
jb_cp_cls_c_632:31510883
748c7209:31497895
cg448276_tst_on:31511779
b1g4i769:31505233
ha629193:31508444
539gd155_c:31506875

```

</details>
